packages/devextreme/js/__internal/ui/html_editor/utils/m_toolbar_helper.ts

```diff
@@ -102,6 +103,63 @@ function getFormatHandlers(module) {
     deleteTable: getTableOperationHandler(module.quill, 'deleteTable'),
     cellProperties: prepareShowFormProperties(module, 'cell'),
     tableProperties: prepareShowFormProperties(module, 'table'),
+    ai: handleAiDropDownSelection(module),
+  };
+}
+
+function handleAiDropDownSelection(module) {
+  return (options): void => {
+    const { command, parentCommand, commandsMap } = options;
+
+    if (command === 'root') {
+      return;
+    }
+
+    const { quill } = module;
+    const selection = quill.getSelection();
+    const hasSelection = selection?.length > 0;
+    const text = hasSelection ? quill.getText(selection) : quill.getText();
+
+    const aiDialogConfig: AiDialogShowPayload = {
+      currentCommand: parentCommand ?? command,
+      currentCommandOption: parentCommand ? command : undefined,
+      text,
+      commandsMap,
+    };
+
+    module.editorInstance.showAiDialog(aiDialogConfig)?.done((resultText, event) => {
+      const insertionMode = event.itemData.id;
+      let insertIndex = 0;
+
+      switch (insertionMode) {
+        case 'replace':
+          insertIndex = hasSelection ? selection.index : 0;
+          quill.deleteText(
+            insertIndex,
+            hasSelection ? selection.length : text.length,
```

Comment:
I'm afraid there can be issues when not only text is included. Please add a point to checklist to test such scenarios and compare with implementation in Blazor